### ISSUE TITLE
API endpoint for UI localization does not work with PHP's built-in server

### ISSUE THREAD

**Issue by NateWr** [pkp/pkp-lib#9478]
**Describe the bug**
The API endpoint invoked to get UI localizations returns a 404 error when running PHP's built-in server. This is likely because the URL ends in `.js` and the server interprets this as a static file, so it bypasses the PHP-based routing.

**To Reproduce**
1. Launch OJS with PHP's built-in server: `php -S 0.0.0.0:8000 -c ../path/to/custom/php.ini` 
2. Open the browser's dev tools and go to the Network tab.
3. Load a page with translations, such as Settings > Website.
4. See the 404 network request to `.../api/v1/_i18n/ui.js?hash=...`
5. See missing translations, for example at Settings > Website > Setup > Highlights

**What application are you using?**
`main`  (3.5 pre-release)

**Additional information**
This is the full URL path to `ui.js`:

```
http://localhost:8000/publicknowledge/api/v1/_i18n/ui.js?hash=a8524194c6679b6d3427de9aec12c69fa9516624
```

This behaviour probably depends on the server configuration, but I suspect a lot of people out there are going to have server configurations that interpret URLs ending in `.js` as static requests.

For completeness, this is my custom `php.ini` file that I use when launching the built-in server:

```
display_errors = Off
error_log = /home/nate/Projects/pkp/_server/error.log
log_errors = On
error_reporting = E_ALL ^ E_STRICT
date.timezone = Europe/London
upload_max_filesize = 20M
post_max_size = 20M
max_execution_time = 10
```

**Comment by NateWr** [comment]
Here is my `.htaccess` file, in case that is effecting things:

```
<IfModule mod_rewrite.c>
  RewriteEngine On
  RewriteCond %{REQUEST_FILENAME} !-d
  RewriteCond %{REQUEST_FILENAME} !-f
  RewriteRule ^(.*)$ index.php/$ [QSA,L]
</IfModule>
```

**Comment by asmecher** [comment]
@NateWr, I don't think the `.htaccess` is supported at all by the PHP built-in server, so it's probably not related.

I ran a couple of tests with a `phpinfo` script, adding `PATH_INFO` URL parameters to it that were "shaped like" real filenames; I didn't find it unwilling to handle things that looked like .js filenames. Can you try that on your side? (Drop a `phpinfo()` script right alongside OJS's `index.php` and try the same URL with just the script name changed.)

**Comment by jardakotesovec** [comment]
So far have not been able to reproduce it. If it turns out that .js extension is causing problems on some configuration we can get rid of it. Browser should care about content type.. not the extension.

**Comment by NateWr** [comment]
On a suggestion from @jardakotesovec, I tested turning `restful_urls` off in the config, and the http request worked. So maybe this is related to `restful_urls` being on.

**Comment by jardakotesovec** [comment]
@asmecher When restful_urls is On, php dev server attempts to load it from disk, because of the extension. I think dropping extension does not cause problems (and it certainly solves this one). Content-type is what matters for browser.

If you agree we can merge it here - https://github.com/pkp/pkp-lib/pull/9486

**Comment by asmecher** [comment]
Hi all -- I've filed this for attention in the PHP repo with details:

https://github.com/php/php-src/issues/12604

Especially since the PHP server is development-only, I'd prefer this was fixed upstream. Having `.` appear in `PATH_INFO` is going to be a useful thing, if not here then elsewhere, and the PHP dev server is clearly out of spec. I'm not sure how much interest they'll have in fixing it, but meanwhile, we can just expect to have to disable `restful_urls` when using it.
